### PR TITLE
[DO-170] adding optional policy creation for ECR repositories

### PR DESCRIPTION
**Pull request by simplyadrian**
@notyost @sapiensantiquus please review

### CODE REVIEW

File: templates/ecr.json.j2 [hunk 1]
```diff
@@ -11,7 +11,16 @@
...
         "RepositoryName" : { "Ref" : "RepositoryName"}
+        {% if ecr_policy is defined %}
+        ,
+        "RepositoryPolicyText" : {{ ecr_policy|to_json }}
+        {% endif %}
       }
     }
+  },
+  "Outputs" : {
+    "RepositoryName" : {
```
Comment (sapiensantiquus) [inline templates/ecr.json.j2]: Maybe RepositoryArn for clarity? I see a lot of roles that have both a name and an ARN.

File: tests/policy.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{
+  "Version": "2008-10-17",
+  "Statement" : [
+    {
+      "Sid" : "CrossAccountPull",
+      "Effect" : "Allow",
+      "Principal" : {
+        "AWS" : [
+          "arn:aws:iam::209983464176:root",
```
Comment (sapiensantiquus) [inline tests/policy.json]: Not sure if we care if these are public?

File: templates/ecr.json.j2 (hunk 1 above)
Comment (simplyadrian) [inline templates/ecr.json.j2]: @sapiensantiquus adding in fix. please recheck.

**Comment by sapiensantiquus** [comment]
👍

**Pull request merged**